trunk/examples/concurrent/pthreads/races/regression/atomic_type_array.c

```diff
@@ -0,0 +1,24 @@
+//#Safe
+
+/*
+ * Author: Frank Schüssele (schuessf@informatik.uni-freiburg.de)
+ * Date: 2024-10-18
+ */
+
+typedef unsigned long pthread_t;
+
+_Atomic int x[10];
```

Comment:
Is this an atomic array of integers, or an array of atomic integers? And do we support this usage?

Reply:
Hmm, sentence 3 in section 6.7.3 of the C standard (N1570) says

> The type modified by the _Atomic qualifier shall not be an array type or a function type.

Is this an issue here?

Reply:
> Is this an atomic array of integers, or an array of atomic integers? And do we support this usage?

This should be an array of atomic integers (`(_Atomic int)[]`) and we support this, therefore I added this regression test.

> Hmm, sentence 3 in section 6.7.3 of the C standard (N1570) says
> 
> > The type modified by the _Atomic qualifier shall not be an array type or a function type.
> 
> Is this an issue here?

No, this means that `_Atomic (int[])` is disallowed and this coincides with the change to `CArray` and `CFunction` (the last parameter is always false, i.e., we never consider an atomic array type).

Reply:
:+1: ok, I was just unsure because you had also added some regression test above where the comment says we would fail. So I wasn't sure if there should be a similar comment here.